Gems/AzQtComponentsForPython/CMakeLists.txt

```diff
@@ -0,0 +1,224 @@
+set(BINDINGS_LIBRARY "azqtpyside")
+
+# Header file containing all the types/classes we're exposing to Python.
+set(wrapped_header ${CMAKE_CURRENT_LIST_DIR}/bindings_azqtcomponents.h)
+
+# Typesystem xml.
+set(typesystem_file ${CMAKE_CURRENT_LIST_DIR}/bindings_azqtcomponents.xml)
+
+# Allow AUTOMOC/AUTOUIC on generated files.
+if(POLICY CMP0071)
+  cmake_policy(SET CMP0071 NEW)
+endif()
+
+project(AzQtComponent_Bindings LANGUAGES CXX)
+set(CMAKE_AUTOMOC ON)
+find_package(Qt5 5.12 REQUIRED COMPONENTS Core Gui Widgets)
+             
+# The library we're creating bindings for.
+set(AZQTCOMPONENTS_LIBRARY "AzQtComponents")
+
+# Check we have a valid version of Python.
+if(NOT python_interpreter)
+    find_program(python_interpreter "python")
+endif()
+
+if(NOT python_interpreter)
+    message(FATAL_ERROR "Error: Failed to locate Python installation.")
+endif()
+
+set(LIB_EXTENSION "cp310-win_amd64")
+
+# Find the paths for the pyside2/shiboken modules.
+ly_get_package_association("pyside2" PYSIDE_PACKAGE_NAME)
+ly_package_get_target_folder(${PYSIDE_PACKAGE_NAME} PYSIDE2_DOWNLOAD_LOCATION)
+set(PYSIDE2_MODULE_DIR ${PYSIDE2_DOWNLOAD_LOCATION}/${PYSIDE_PACKAGE_NAME})
+
+# Find the qt package location.
+ly_get_package_association("Qt" QT_PACKAGE_NAME)
+ly_package_get_target_folder(${QT_PACKAGE_NAME} QT_DOWNLOAD_LOCATION)
+set(QT_MODULE_DIR ${QT_DOWNLOAD_LOCATION}/${QT_PACKAGE_NAME}/qt)
+
+set(PYSIDE2_PATH "${PYSIDE2_MODULE_DIR}/pyside2")
+set(INCLUDE_DIR "${PYSIDE2_PATH}/include")
+set(PYSIDE2_SHARE_PATH "${PYSIDE2_PATH}/share/PySide2")
+set(PYSIDE2_TYPESYSTEM_PATH "${PYSIDE2_SHARE_PATH}/typesystems")
+set(PYSIDE2_BIN_PATH "${PYSIDE2_PATH}/bin")
+set(PYSIDE2_INCLUDE_DIR ${INCLUDE_DIR}/PySide2)
+set(SHIBOKEN_GENERATOR_EXE_PATH "${PYSIDE2_BIN_PATH}/shiboken2${CMAKE_EXECUTABLE_SUFFIX}") 
+
+if(NOT EXISTS ${SHIBOKEN_GENERATOR_EXE_PATH})
+    message(FATAL_ERROR "Shiboken executable not found at path: ${SHIBOKEN_GENERATOR_EXE_PATH}")
+endif()
+  
+set(SHIBOKEN_MODULE_PATH "${PYSIDE2_MODULE_DIR}/pyside2/lib/site-packages/shiboken2")
+
+set(SHIBOKEN_INCLUDE_DIR "${INCLUDE_DIR}/shiboken2")
+set(SHIBOKEN_SHARED_LIBRARIES "${SHIBOKEN_MODULE_PATH}/shiboken2.abi3.lib")
+set(SHIBOKEN_DLL "${SHIBOKEN_MODULE_PATH}/shiboken2.abi3.dll")
+set(SHIBOKEN_SHARED_LIBRARIES_D "${SHIBOKEN_MODULE_PATH}/shiboken2_d.${LIB_EXTENSION}.lib")
+set(SHIBOKEN_DLL_D "${SHIBOKEN_MODULE_PATH}/shiboken2_d.${LIB_EXTENSION}.dll")
+
+set(PYSIDE2_LIB_DIR ${PYSIDE2_PATH}/lib/site-packages/PySide2)
+
+set(PYSIDE2_SHARED_LIBRARIES "${PYSIDE2_LIB_DIR}/pyside2.abi3.lib")
+set(PYSIDE2_SHARED_LIBRARIES_D "${PYSIDE2_LIB_DIR}/pyside2_d.${LIB_EXTENSION}.lib")
+
+set(PYSIDE2_DLL "${PYSIDE2_LIB_DIR}/pyside2.abi3.dll")
+set(PYSIDE2_DLL_D "${PYSIDE2_LIB_DIR}/pyside2_d.${LIB_EXTENSION}.dll")
+  
+set(CMAKE_SKIP_BUILD_RPATH FALSE)
+set(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
+set(CMAKE_INSTALL_RPATH ${SHIBOKEN_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR})
+set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
```

Comment:
We have a [pyside2](https://github.com/o3de/3p-package-source/blob/main/package-system/pyside2/Findpyside2.cmake.Windows) 3rdParty package, that comes with the appropriate package association and cmake find scripts that can be used on Windows and Linux.

This CMakeLists.txt will not compile on Linux.